syntax-sloc/lang-file-sloc.rkt

```diff
@@ -6,20 +6,36 @@
 
 (provide lang-file-sloc)
 
-(require "syntax-sloc.rkt"
-         typed/syntax-sloc/read-lang-file
-         syntax-sloc/syntax-sloc)
+(require racket/set
+         syntax-sloc/syntax-sloc
+         typed/syntax-sloc/read-lang-file)
 
 (module+ test
   (require typed/rackunit
            racket/runtime-path))
 
-(: lang-file-sloc : Path-String [#:use-stx? Stx-Pred] -> Natural)
-(define (lang-file-sloc path-string #:use-stx? [use-stx? #f])
-  (syntax-sloc (read-lang-file path-string) #:use-stx? use-stx?))
+(: lang-file-sloc : Path-String [#:include-stx? Stx-Pred] -> Natural)
+(define (lang-file-sloc path-string #:include-stx? [include-stx? #f])
+  (define raw-stx (read-lang-file path-string))
+  (define the-src (syntax-source raw-stx))
+  (define exp-stx
+    (parameterize ([current-namespace (make-base-namespace)])
+      (expand raw-stx)))
+  (define stx-to-count
+    (if include-stx?
+      (filter/stx exp-stx include-stx?)
+      (list exp-stx)))
+  (define (id/the-src? (stx : (Syntaxof Any))) : Boolean
+    (and (identifier? stx) (equal? the-src (syntax-source stx))))
+  (define lines-from-identifiers
+    (for/fold : (Setof Natural)
+              ([acc : (Setof Natural) (set)])
+              ([stx (in-list stx-to-count)])
+      (source-lines/filter stx id/the-src? acc)))
+  (set-count lines-from-identifiers))
```

Comment:
Instead of using `filter/stx` and then using `source-lines/filter`, have you tried using `(syntax-sloc/filter stx include-stx? #:count-line? (src-matches? the-src))` or something? You'll have to re-do this based on my `count-line-pred` branch for that.